sklearn/ensemble/tests/test_stacking.py

```diff
@@ -646,6 +654,117 @@ def fit(self, X, y):
         stacker.n_features_in_
 
 
+@pytest.mark.parametrize(
+    "stack_method", ["auto", "predict", "predict_proba", "decision_function"]
+)
+@pytest.mark.parametrize("passthrough", [False, True])
+def test_stacking_classifier_multilabel(stack_method, passthrough):
+    """Check the behaviour of the `_concatenate_prediction` utils
+    when `y` is multilabel.
+    """
+    X_train, X_test, y_train, y_test = train_test_split(
+        X_multilabel, y_multilabel, stratify=y_multilabel, random_state=42
+    )
+
+    if stack_method == "decision_function":
+        estimators = [
+            ("ridge_1", RidgeClassifier()),
+            ("ridge_2", RidgeClassifier()),
+        ]
+    else:
+        estimators = [
+            ("mlp", MLPClassifier()),
+            ("knnc", RandomForestClassifier()),
+            ("dcs", DummyClassifier(strategy="stratified", random_state=42)),
+            ("dcu", DummyClassifier(strategy="uniform", random_state=42)),
+        ]
+
+    final_estimator = KNeighborsClassifier()
+
+    clf = StackingClassifier(
+        estimators=estimators,
+        final_estimator=final_estimator,
+        passthrough=passthrough,
+        stack_method=stack_method,
+    )
+    y_train_before_fit = y_train.copy()
+    clf.fit(X_train, y_train)
+    assert_array_equal(y_train_before_fit, y_train)
+
+    y_pred = clf.predict(X_test)
+    assert y_pred.shape == y_test.shape
+
+    y_proba = clf.predict_proba(X_test)
+
+    X_trans = clf.transform(X_test)
+    expected_n_features = len(estimators)
+
+    assert y_proba.shape == y_test.shape
+    assert len(clf.classes_) == y_test.shape[1]
+    expected_n_features *= y_test.shape[1]
+
+    if passthrough:
+        X_test_last_cols = X_trans[:, -X_test.shape[1] :]
+        assert_array_equal(X_test, X_test_last_cols)
+        expected_n_features += X_test.shape[1]
+
+    assert X_trans.shape[1] == expected_n_features
+
+
+@pytest.mark.parametrize(
+    "stack_method", ["auto", "predict", "predict_proba", "decision_function"]
+)
+@pytest.mark.parametrize("passthrough", [False, True])
+def test_stacking_classifier_binary(stack_method, passthrough):
```

Comment:
We already covered the binary case in some tests above.
We could drop this test.